contracts/proposers/BurnAuction.sol

```diff
@@ -66,6 +101,26 @@ contract BurnAuction is Chooser {
         return auction[_currentSlot].coordinator;
     }
 
+    function withdrawDonation() external {
+        require(
+            deposits[donationAddress] != 0,
+            "BurnAuction, withdrawDonation: donation deposit is zero"
+        );
+        uint256 donationAmount = deposits[donationAddress];
+        deposits[donationAddress] = 0;
+        donationAddress.transfer(donationAmount);
+    }
+
+    function witdraw(uint256 amount) external {
+        address payable claimer = msg.sender;
+        require(
+            deposits[claimer] >= amount,
+            "BurnAuction, withdraw: insufficient deposit amount for withdraw"
+        );
+        claimer.transfer(amount);
+        updateBalance(claimer, 0, amount);
```

Comment:
We should update before a transfer to prevent a reentrance.
```suggestion
        updateBalance(claimer, 0, amount);
        claimer.transfer(amount);
```